### ISSUE TITLE
Webpack HMR: Uncaught TypeError: Cannot read property 'indexOf' of undefined

### ISSUE THREAD

**Issue by wzt3309** [webpack/webpack#6983]
# my webpack.config.js
```js
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const CleanWebpackPlugin = require('clean-webpack-plugin');
const webpack = require('webpack');

module.exports = {
    entry: {
        app: './src/index.js'
    },
    devtool: 'inline-source-map',
    devServer: {
        contentBase: './dist',
        hot: true
    },
    output: {
        filename: '[name].bundle.js',
        path: path.resolve(__dirname, 'dist')
    },
    plugins: [
        new CleanWebpackPlugin(['dist']),
        new HtmlWebpackPlugin({
            title: 'Hot Module Replacement'
        }),
        new webpack.NamedModulesPlugin(),
        new webpack.HotModuleReplacementPlugin()
    ]
};
```
# my package.json

```json
{
  "name": "webpack-demo",
  "version": "1.0.0",
  "description": "",
  "private": true,
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "watch": "webpack --watch",
    "start": "webpack-dev-server --open",
    "server": "node server.js",
    "build": "webpack"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "clean-webpack-plugin": "^0.1.19",
    "css-loader": "^0.28.11",
    "express": "^4.16.3",
    "file-loader": "^1.1.11",
    "html-webpack-plugin": "^3.2.0",
    "style-loader": "^0.20.3",
    "w": "^1.1.0",
    "webpack": "^4.5.0",
    "webpack-cli": "^2.0.14",
    "webpack-dev-middleware": "^3.1.2",
    "webpack-dev-server": "^3.1.2"
  },
  "dependencies": {
    "lodash": "^4.17.5"
  }
}

```

when i run `npm start`, everything is ok, here is chrome console output:
```
[HMR] Waiting for update signal from WDS...              app.bundle.js:9 
[WDS] Hot Module Replacement enabled.                  app.bundle.js:9 
```

and then I do what [https://webpack.js.org/guides/hot-module-replacement/](https://webpack.js.org/guides/hot-module-replacement/) said, change `./src/print.js`

but some errors happen. chrome console ouput:
```
 [WDS] App hot update...            client:223
Uncaught TypeError: Cannot read property 'indexOf' of undefined         dev-server.js:9 
    at o (dev-server.js:9)
    at r.<anonymous> (dev-server.js:53)
    at r.emit (events.js:81)
    at _ (client:226)
    at Object.warnings (client:158)
    at E.i.onmessage (socket.js:41)
    at E.n.dispatchEvent (sockjs.js:170)
    at sockjs.js:883
    at Array.forEach (<anonymous>)
    at E._transportMessage (sockjs.js:881)
```

and HRM is not work at all

here is my index.js and print.js

```js
// index.js
import _ from 'lodash';
import printMe from './print';

function component() {
    var element = document.createElement('div');
    var btn = document.createElement('button');

    element.innerHTML = _.join(['Hello', 'webpack'], ' ');

    btn.innerHTML = 'Click me and check the console';
    btn.onclick = printMe;

    element.appendChild(btn);

    return element;
  }
  
  let element = component();
  document.body.appendChild(element);

  // don't need to flush all when do some change
  if (module.hot) {
    module.hot.accept('./print.js', function() {
      console.log('Accepting the updated printMe module!'); 
      document.body.removeChild(element);
      element = component(); // Re-render the "component" to update the click handler
      document.body.appendChild(element);
    })
  }

```
```js
// print.js
export default function printMe() {
    // i change this line to test HRM
    console.log('Updating prinst.js...');
}
```

Pls help me

**Comment by wzt3309** [comment]
I solved this problem in a way which i don't like :(

# step1 change the version of webpack-dev-sever
`npm uninstall webpack-dev-server`
`npm install webpack-dev-server@2.9.7`
```json
{
  "name": "webpack-demo",
  "version": "1.0.0",
  "description": "",
  "private": true,
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "watch": "webpack --watch",
    "start": "webpack-dev-server --open",
    "server": "node server.js",
    "build": "webpack"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "clean-webpack-plugin": "^0.1.19",
    "html-webpack-plugin": "^3.2.0",
    "webpack": "^4.5.0",
    "webpack-cli": "^2.0.14",
    "webpack-dev-server": "^2.9.7"
  },
  "dependencies": {
    "lodash": "^4.17.5"
  }
}
```
# step2
create a server.js 
```js
const webpackDevServer = require('webpack-dev-server');
const webpack = require('webpack');

const config = require('./webpack.config.js');
const options = {
  contentBase: './dist',
  hot: true,
  host: 'localhost'
};

webpackDevServer.addDevServerEntrypoints(config, options);
const compiler = webpack(config);
const server = new webpackDevServer(compiler, options);

server.listen(5000, 'localhost', () => {
  console.log('dev server listening on port 5000');
});
```
# step3
`npm run server`

every thing is done now

Any one has the best way? because i want to use the latest version of webpeck-dev-server

**Comment by bayunjiang** [comment]
It works on 3.1.1

**Comment by ooflorent** [comment]
`webpack-dev-server` is going to maintenance mode. You should try `webpack-serve`. Moreover, it looks like the issue is related to WDS, not webpack. You could a new issue in its repositiory.

**Comment by livemixlove** [comment]
For others arriving here via google,  webpack-dev-server continues to be the supported/current tool of choice.   https://webpack.js.org/configuration/dev-server/

webpack-serve is deprecated.

**Comment by pcarballeda** [comment]
I have the same problem after updating to the latest version (3.3.1)

**Comment by evilebottnawi** [comment]
@pcarballeda it is fixed in `3.3.1`, you still use old version
